src/dag-builder/index.js

```diff
@@ -33,10 +33,18 @@ async function * dagBuilder (source, ipld, options) {
       const chunker = createChunker(options.chunker, validateChunks(source), options.chunkerOptions)
 
       // item is a file
-      yield fileBuilder(entry, chunker, ipld, options)
+      yield fileBuilder(entry, chunker, ipld, {
+        ...options,
+        mode: entry.mode,
+        mtime: entry.mtime
+      })
     } else {
       // item is a directory
-      yield dirBuilder(entry, ipld, options)
+      yield dirBuilder(entry, ipld, {
+        ...options,
+        mode: entry.mode,
+        mtime: entry.mtime
```

Comment:
Is it correct that these are not used by the `dirBuilder` but are passed to any files linked to from the directory?

Reply:
No, just the jetlag talking